### PR TITLE
Create nightly matrix of pulpcore versions

### PR DESCRIPTION
**Pull request by daviddavis**
[noissue]

### CODE REVIEW

**Comment by daviddavis** [comment]
Here's an example run: https://github.com/daviddavis/pulp-cli/runs/1602388622

It looks like the pulpexporter test is failing against pulpcore 3.6.

**Comment by daviddavis** [comment]
This may be a terrible idea as I am not sure how we'll handle the case where new functionality depends on new pulpcore versions.

**Comment by daviddavis** [comment]
Actually, I think we could use $PULPCORE_VERSION in our tests.

**Comment by mdellweg** [comment]
In the cli you can guard with `pulp_ctx.has_plugin("pulpcore", min_version="...")`

File: .ci/run_container.sh [hunk 1]
```diff
@@ -12,7 +13,13 @@ then
   fi
 fi
 
-"${CONTAINER_RUNTIME}" run --rm --detach --name "pulp" --volume "${BASEPATH}/settings:/etc/pulp" --publish "8080:80" pulp/pulp-fedora31
+if [ -n "$PULPCORE_VERSION" ]
+then
+  "${CONTAINER_RUNTIME}" build -f "${BASEPATH}/Containerfile" -t pulp --build-arg PULPCORE_VERSION="==$PULPCORE_VERSION" .
+  IMAGE="pulp"
+fi
```
Comment (mdellweg) [inline .ci/run_container.sh:20]: It would be really nice to have those images available from pulp-oci-images

Comment (daviddavis) [reply .ci/run_container.sh:20]: Here we go: https://github.com/pulp/pulp-oci-images/pull/37

File: .github/workflows/nightly.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  CONTAINER_RUNTIME: 'docker'
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        image_tag:
+          - "latest"
```
Comment (mdellweg) [inline .github/workflows/nightly.yml:20]: Do we want 3.9 here also, or are you assuming that "latest" is the same?

File: .github/workflows/nightly.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+          - "3.8"
+          - "3.7"
+          - "3.6"
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.7
```
Comment (mdellweg) [inline .github/workflows/nightly.yml:29]: Not sure, it is a good idea, to use a python version not covered by the PR workflows.

**Review by mdellweg — APPROVED** [review]
Left two notes, but nothing i could not be convinced of.
This is a great piece of work, and it will help us with the version specific workarounds.

File: .github/workflows/nightly.yml (hunk 2 above)
Comment (daviddavis) [reply .github/workflows/nightly.yml:20]: It should be the same. In fact, there's no 3.9 container at the moment. I'm open to the idea of having one though.

**Pull request merged**